core/src/test/java/com/graphhopper/routing/util/BikeFlagEncoderTest.java

```diff
@@ -68,23 +68,40 @@ public void testGetSpeed()
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("bicycle", "yes");
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
         way.setTag("highway", "footway");
         way.setTag("surface", "paved");
         way.setTag("bicycle", "yes");
 
-        assertEquals(6, encoder.getSpeed(way));
+        assertEquals(PUSHING_SECTION_SPEED * 2, encoder.getSpeed(way));
         assertPriority(UNCHANGED.getValue(), way);
 
         way.clearTags();
+        way.setTag("highway", "cycleway");
+        assertEquals(18, encoder.getSpeed(way));
+        assertPriority(VERY_NICE.getValue(), way);
+        int cyclewayspeed = encoder.getSpeed(way);
```

Comment:
and a check is missing where highway=path without another tag resulting in the path speed=12 ?